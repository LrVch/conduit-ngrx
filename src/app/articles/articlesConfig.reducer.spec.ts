import * as fromArticlesConfig from './articlesConfig.reducer';
import {
  SetTypeOfFeed,
  SetTag,
  SetAuthor,
  SetOffset,
  SetLimit,
  SetPageIndex,
  SetFavorited,
  ResetConfig
} from './articles.actions';

describe('ArticleConfig Reducer', () => {
  describe('State changes', () => {
    const initialArticleConfigState: fromArticlesConfig.ArticlesConfigState =
      fromArticlesConfig.initialState;

    it('should have an initial state', () => {
      const state = fromArticlesConfig.articlesConfigReducer(
        initialArticleConfigState,
        { type: '@@init' } as any
      );

      expect(state).toBe(initialArticleConfigState);
    });

    it('should type as "all" on "SetTag"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetTypeOfFeed({ type: 'all' });

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });

    it('should tag as "color" on "SetTag"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetTag({ tag: 'color' });

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });

    it('should author as "author" on "SetAuthor"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetAuthor({ author: 'author' });

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });

    it('should offset as "20" on "SetOffset"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetOffset({ offset: 20 });

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });

    it('should limit as "20" on "SetLimit"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetLimit({ limit: 20 });

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });

    it('should pageIndex as "20" on "SetPageIndex"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetPageIndex({ pageIndex: 20 });

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });

    it('should favorited as "favorited" on "SetFavorited"', () => {
      const init = { type: '@@init' } as any;
      const action = new SetFavorited({ favorited: 'favorited' });

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });

    it('should reset config on "ResetConfig"', () => {
      const init = { type: '@@init' } as any;
      const action = new ResetConfig();

      const state = [init, action].reduce(
        fromArticlesConfig.articlesConfigReducer,
        initialArticleConfigState
      );

      expect(state).toMatchSnapshot();
    });
  });
});
