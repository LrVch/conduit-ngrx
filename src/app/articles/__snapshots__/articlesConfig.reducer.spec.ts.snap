// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleConfig Reducer State changes should author as "author" on "SetAuthor" 1`] = `
Object {
  "filters": Object {
    "author": "author",
    "favorited": "",
    "limit": 10,
    "offset": 0,
    "pageIndex": 0,
    "tag": "",
  },
  "type": "all",
}
`;

exports[`ArticleConfig Reducer State changes should favorited as "favorited" on "SetFavorited" 1`] = `
Object {
  "filters": Object {
    "author": "",
    "favorited": "favorited",
    "limit": 10,
    "offset": 0,
    "pageIndex": 0,
    "tag": "",
  },
  "type": "all",
}
`;

exports[`ArticleConfig Reducer State changes should limit as "20" on "SetLimit" 1`] = `
Object {
  "filters": Object {
    "author": "",
    "favorited": "",
    "limit": 20,
    "offset": 0,
    "pageIndex": 0,
    "tag": "",
  },
  "type": "all",
}
`;

exports[`ArticleConfig Reducer State changes should offset as "20" on "SetOffset" 1`] = `
Object {
  "filters": Object {
    "author": "",
    "favorited": "",
    "limit": 10,
    "offset": 20,
    "pageIndex": 0,
    "tag": "",
  },
  "type": "all",
}
`;

exports[`ArticleConfig Reducer State changes should pageIndex as "20" on "SetPageIndex" 1`] = `
Object {
  "filters": Object {
    "author": "",
    "favorited": "",
    "limit": 10,
    "offset": 0,
    "pageIndex": 20,
    "tag": "",
  },
  "type": "all",
}
`;

exports[`ArticleConfig Reducer State changes should reset config on "ResetConfig" 1`] = `
Object {
  "filters": Object {
    "author": "",
    "favorited": "",
    "limit": 10,
    "offset": 0,
    "pageIndex": 0,
    "tag": "",
  },
  "type": "all",
}
`;

exports[`ArticleConfig Reducer State changes should tag as "color" on "SetTag" 1`] = `
Object {
  "filters": Object {
    "author": "",
    "favorited": "",
    "limit": 10,
    "offset": 0,
    "pageIndex": 0,
    "tag": "color",
  },
  "type": "all",
}
`;

exports[`ArticleConfig Reducer State changes should type as "all" on "SetTag" 1`] = `
Object {
  "filters": Object {
    "author": "",
    "favorited": "",
    "limit": 10,
    "offset": 0,
    "pageIndex": 0,
    "tag": "",
  },
  "type": "all",
}
`;
